lms/djangoapps/courseware/views/index.py

```diff
@@ -250,7 +237,7 @@ def render(self, request):
                 'email_opt_in': False,
             })
 
-            allow_anonymous = allow_public_access(self.course, [COURSE_VISIBILITY_PUBLIC])
+            allow_anonymous = check_public_access(self.course, [COURSE_VISIBILITY_PUBLIC])
```

Comment:
As mentioned earlier, `check_public_access` now returns an AccessResponse, which can be evaluated as a boolean.  I looked through every call to `check_public_access` and don't see anywhere that this is a problem.  Most places are these `allow_anonymous` variables which get used in boolean logic.  There's _one_ place where it's set in a context and passed to a template.  In that case, it's still just evaluated as a boolean in python code in the template, so I think it's all fine.